psycopg/psycopg/cursor.py

```diff
@@ -483,11 +471,34 @@ def _set_current_result(self, i: int, format: Optional[Format] = None) -> None:
 
     def _set_results_from_pipeline(self, results: List["PGresult"]) -> None:
         self._check_results(results)
-        if not self._results:
-            self._results = results
-            self._set_current_result(0)
-        else:
+        first_batch = not self._results
+
+        if self._execmany_returning is None:
+            # Received from execute()
+            # TODO: bug we also end up here on executemany() if run from inside
+            # a pipeline block. This causes a wrong rowcount. As it isn't so
+            # serious, currently leaving it this way.
+            first_batch = not self._results
```

Comment:
This line is redundant with l. 474 above, no?

Reply:
Yes it is, thank you :)